spyder/api/preferences.py

```diff
@@ -124,6 +128,10 @@ def aggregate_sections_partials(self, opts):
         """Aggregate options by sections in order to notify observers."""
         to_update = {}
         for opt in opts:
+            if isinstance(opt, tuple):
+                if len(opt) == 2 and opt[0] is None:
+                    opt = opt[1]
```

Comment:
Please add a comment above this new `if` to understand why it's necessary. It seems to be here for a very specific use case.